From: Hans-Joachim Picht <hpicht@redhat.com>
Date: Mon, 10 Dec 2007 14:03:46 +0100
Subject: [s390] zfcp: Reduce flood on hba trace
Message-id: 20071210130346.GC29540@redhat.com
O-Subject: [RHEL5 U2 PATCH 3/8] s390 - zfcp: Reduce flood on hba trace
Bugzilla: 415951
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>

Description
============

The debug trace s390dbf/zfcp*hba contains entries with resp/qual for all
successfully processed FSF commands.

Errors which would be of interest are hard to find or are already overwritten.
The protocol to the adapter changed: A field that used to only describe errors
now contains measurement data by default. This measurement data triggers the
tracing of all normal responses.

The fix is to simply remove the "qual" tracing: The responses with an
interesting status are also traced as "ferr" or "perr" and all responses can be
traced as "norm" with a higher trace level.

Bugzilla
=========

BZ 415951
https://bugzilla.redhat.com/show_bug.cgi?id=415951

Upstream status of the patch:
=============================
The patch is upstream in the scsi-misc tree (commit
be376c7b2d5e6ba400645b5938025358ddc14ec1).

Test status:
============
Kernel with patch was built and successfully tested

Please ACK.

With best regards,

Hans

 drivers/s390/scsi/zfcp_dbf.c |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/drivers/s390/scsi/zfcp_dbf.c b/drivers/s390/scsi/zfcp_dbf.c
index 0aa3b1a..6a8a97e 100644
--- a/drivers/s390/scsi/zfcp_dbf.c
+++ b/drivers/s390/scsi/zfcp_dbf.c
@@ -161,12 +161,6 @@ inline void zfcp_hba_dbf_event_fsf_response(struct zfcp_fsf_req *fsf_req)
 		   (fsf_req->fsf_command == FSF_QTCB_OPEN_LUN)) {
 		strncpy(rec->tag2, "open", ZFCP_DBF_TAG_SIZE);
 		level = 4;
-	} else if ((prot_status_qual->doubleword[0] != 0) ||
-		   (prot_status_qual->doubleword[1] != 0) ||
-		   (fsf_status_qual->doubleword[0] != 0) ||
-		   (fsf_status_qual->doubleword[1] != 0)) {
-		strncpy(rec->tag2, "qual", ZFCP_DBF_TAG_SIZE);
-		level = 3;
 	} else {
 		strncpy(rec->tag2, "norm", ZFCP_DBF_TAG_SIZE);
 		level = 6;
